nsxt/policy_utils.go

```diff
@@ -180,6 +196,39 @@ func nsxtDomainResourceImporter(d *schema.ResourceData, m interface{}) ([]*schem
 	return []*schema.ResourceData{d}, nil
 }
 
+func nsxtPolicyPathResourceImporter(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	rd, err := nsxtPolicyPathResourceImporterHelper(d, m)
+	if errors.Is(err, ErrNotAPolicyPath) {
+		return rd, nil
+	} else if err != nil {
+		return rd, err
+	}
+	return rd, nil
+}
+
+func nsxtPolicyPathResourceImporterHelper(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	importID := d.Id()
+	if isPolicyPath(importID) {
+		pathSegs := strings.Split(importID, "/")
+		if len(pathSegs) < 4 {
+			return nil, fmt.Errorf("invalid policy path %s", importID)
+		} else if strings.Contains(pathSegs[1], "infra") {
+			d.SetId(pathSegs[len(pathSegs)-1])
+		} else if pathSegs[1] == "orgs" && pathSegs[3] == "projects" {
+			// pathSegs[2] should contain the organization. Once we support multiple organization, it should be
+			// assigned into the context as well
+			contexts := make([]interface{}, 1)
+			ctxMap := make(map[string]interface{})
+			ctxMap["project_id"] = pathSegs[4]
```

Comment:
can we get a segmentation fault here if user provides bad path?

Reply:
So maybe we should check for len(pathSegs < 5) for project case